### PR TITLE
[GPU] Blocked layouts support for ExperimentalDetectronGenerateProposalsSingleImage

### PR DESCRIPTION
**Pull request by okhovan**
- added blocked layouts support;
- added unit tests for blocked layouts.

### Tickets:
 - 89241

### CODE REVIEW

File: src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp [hunk 1]
```diff
@@ -151,12 +246,20 @@ struct experimental_detectron_generate_proposals_single_image_test
...
 
-        const auto rois = outputs.at(edgpsi_id).get_memory();
+        const auto rois = outputs.at(reorder_result_id).get_memory();
 
         const cldnn::mem_lock<T> rois_ptr(rois, get_test_stream());
         ASSERT_EQ(rois_ptr.size(), param.post_nms_count * 4);
 
-        const cldnn::mem_lock<T> roi_scores_ptr(output_roi_scores, get_test_stream());
+        cldnn::topology reorder_topology;
+        reorder_topology.add(input_layout("scores", rois_scores_layout));
```
Comment (OlehKravchyshyn) [inline src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp:255]: can't all those transformations be done by the previous network, mean by network.execute()

Comment (okhovan) [reply src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp:255]: No. Plugin does not support multiple outputs natively, and scores and rois_num are, technically, inputs of the primitive.
From network prospective, they are not dependent on GenerateProposals layer, so, due to optimization their reorders would happen in the very beginning, before GenerateProposals. That's why we need separate network to reorder additional outputs.

**Comment by wilson-seok** [comment]
No performance regression during int8 models.

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edgpsi/experimental_detectron_generate_proposals_single_image_kernel_ref.cpp [hunk 2]
```diff
@@ -15,10 +15,11 @@ ParamsKey ExperimentalDetectronGenerateProposalsSingleImageRef::GetSupportedKey(
     k.EnableInputDataType(Datatype::F32);
     k.EnableOutputDataType(Datatype::F16);
     k.EnableOutputDataType(Datatype::F32);
-    k.EnableInputLayout(DataLayout::bfyx);
-    k.EnableOutputLayout(DataLayout::bfyx);
+    k.EnableAllInputLayout();
```
Comment (wilson-seok) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edgpsi/experimental_detectron_generate_proposals_single_image_kernel_ref.cpp:18]: All layout is good for this op? Others have explicitly blocked formats usually.

**Review by wilson-seok — APPROVED** [review]

**Comment by yeonbok** [comment]
This is not updating the whitelists in the program.cpp, to enable blocked formats for the network having this operation. So, the network will still not use blocked formats..

File: src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp [hunk 3]
```diff
@@ -177,76 +280,30 @@ struct experimental_detectron_generate_proposals_single_image_test
...
-    };
-    return params;
-}
-
 INSTANTIATE_TEST_SUITE_P(
         experimental_detectron_generate_proposals_single_image_gpu_test,
         experimental_detectron_generate_proposals_single_image_test_f32,
-        ::testing::ValuesIn(getExperimentalDetectronGenerateProposalsSingleImageParams<float>()));
+        ::testing::Combine(
+                ::testing::ValuesIn(getExperimentalDetectronGenerateProposalsSingleImageParams<float>()),
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp:290]: Please not to use Combine for creating test as much as possible. Instead, please try to pick meaningful test instances, i.e., F > 16 is preferable to test fsv16 and B>16 is preferable to test bsv32.

Comment (okhovan) [reply src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp:295]: This operation has nothing to do with batches and features - it processes single-image inputs (see https://docs.openvino.ai/latest/openvino_docs_ops_detection_ExperimentalDetectronGenerateProposalsSingleImage_6.html ).

Regarding number of tests - agree, reduced it in last commit. 
Also added the operation to program.cpp.

**Review by wilson-seok — APPROVED** [review]

**Comment by wilson-seok** [comment]
@okhovan Please resolve conflicts.

**Review by jade-cho — APPROVED** [review]

**Pull request merged**